keycloak.d.ts

```diff
@@ -0,0 +1,351 @@
+     *
+     * The direct grant API must be enabled for the configured realm
+     * for this method to work. This function ostensibly provides a
+     * non-interactive, programatic way to login to a Keycloak realm.
+     *
+     * @param {String} username The username.
+     * @param {String} password The cleartext password.
+     */
+    obtainDirectly(username: string, password: string): Promise<Grant>
+
+    /**
+     * Obtain a grant from a previous interactive login which results in a code.
+     *
+     * This is typically used by servers which receive the code through a
+     * redirect_uri when sending a user to Keycloak for an interactive login.
+     *
+     * An optional session ID and host may be provided if there is desire for
+     * Keycloak to be aware of this information.  They may be used by Keycloak
+     * when session invalidation is triggered from the Keycloak console itself
+     * during its postbacks to `/k_logout` on the server.
+     *
+     * @param {String} code The code from a successful login redirected from Keycloak.
+     * @param {String} sessionId Optional opaque session-id.
+     * @param {String} sessionHost Optional session host for targetted Keycloak console post-backs.
+     */
+    obtainFromCode(code: string, sessionid?: string, sessionHost?: string, callback?: (err: Error, grant: Grant) => void): Promise<Grant>
+
+
+    /**
+     * Obtain a service account grant.
+     * Client option 'Service Accounts Enabled' needs to be on.
+     *
+     * This method returns or promise or may optionally take a callback function.
+     *
+     * @param {Function} callback Optional callback, if not using promises.
+     */
+    obtainFromClientCredentials (callback?: (err: Error, grant: Grant) => void, scopeParam?: string): Promise<Grant>
+
+    /**
+     * Ensure that a grant is *fresh*, refreshing if required & possible.
+     *
+     * If the access_token is not expired, the grant is left untouched.
+     *
+     * If the access_token is expired, and a refresh_token is available,
+     * the grant is refreshed, in place (no new object is created),
+     * and returned.
+     *
+     * If the access_token is expired and no refresh_token is available,
+     * an error is provided.
+     *
+     * @param {Grant} grant The grant object to ensure freshness of
+     */
+    ensureFreshness (grant: Grant): Promise<Grant>
+
+    /**
+     * Perform live validation of an `access_token` against the Keycloak server.
+     *
+     * @param {Token|String} token The token to validate.
+     * @param {Function} callback Callback function if not using promises.
+     *
+     * @return {boolean} `false` if the token is invalid, or the same token if valid.
+     */
+    validateAccessToken<T extends Token|string>(token: T): Promise<false|T>
+
+    /**
+     * Returns a user info JSON Object
+     * @param {Token|String} token
+     */
+    userInfo<T extends Token|string, C extends StandardClaims>(token: T): Promise<C>
```

Comment:
Sorry to comment on a closed PR, but this line, more specifically the mention of `StandardClaims` prevents me to compile this. As far as I see, `StandardClaims` is not defined anywhere.

Reply:
If you just remove the "extends StandardClaims", is that good enough for you?

Reply:
If so, we can just do a quick fix for it.

Reply:
I just want to point out that if you just remove `extends StandardClaims`, then you're basically saying "this will return whatever the caller wants it to return" and that's probably less than ideal. Maybe good for a quick fix but I wouldn't leave it in the long term.

For the record, there was a TODO in the original PR that got lost when it was turned into this one https://github.com/keycloak/keycloak-nodejs-connect/pull/123/files#r194166591

Reply:
Yes.  I don't have time to work out what StandardClaims needs to be.  If someone in the community (you?) wants to improve this file, please let me know.

So as long as my earlier suggestion will suffice for now, I can do a quick PR.  Have you tried it out just removing "extends StandardClaims"?

Reply:
> If you just remove the "extends StandardClaims", is that good enough for you?

Yep, then it compiles without any error.

Reply:
OK.  Preliminary fix is now merged: https://github.com/keycloak/keycloak-nodejs-connect/pull/210

Reply:
Great, thanks!